api/sip/client.py

```diff
@@ -283,6 +286,7 @@ def __init__(
         use_ssl=False,
         ssl_cert=None,
         ssl_key=None,
+        ssl_contexts: Callable[[_SSLMethod], ssl.SSLContext] = ssl.SSLContext,
```

Comment:
I disagree here, I think its preferable to accurately annotate functions, even if we have to use private members to do it. Part of the reason for trying to accurately annotate the types in our code-base is so that we know when things change. 

If things change upstream when we update, having accurate type annotations here should make it obvious right?